### ISSUE TITLE
[4.x]: Asset indexer from symlink uses wrong path

### ISSUE THREAD

**Issue by ThomasDeMarez** [craftcms/cms#12423]
### Description
When running the `craft index-assets/one documents` command, Craft uses the full path of the referenced symlink to index assets. This results in full folder structure being created and the asset not being accessible.

### Steps to reproduce
1. Create a symlink from `/web/volumes` -> `/shared/web/volumes`
2. Create a new filesystem `@webroot/volumes/documents` 
3. Link filesystem with Volume 'documents'
4. Manually place file(s) in the `/web/volumes/documents` folder
5. Run the `craft index-assets/one documents` command

### Expected behavior
Assets are imported with the relative folders.

### Actual behavior
All folders from the root of the disk are imported, Asset Entries are created but do not link to a physical asset on disk.

<img width="350" alt="Screenshot 2022-12-06 at 16 07 01" src="https://user-images.githubusercontent.com/10974106/205948651-adc71572-1e29-4cc5-940d-ae25473522e8.png">

### Craft CMS version

4.3.1

### PHP version

8.1.12

### Installed plugins and versions

-

**Comment by brianjhanson** [comment]
I'm having trouble replicating this behavior. Your replication steps are wonderful, so I feel like I'm likely missing something.

What does your folder structure look like? From the screenshots, it looks like Craft is behaving a little like you've wound up with nested asset volumes. 

Below is how I have things set up, is there anything I'm doing that doesn't look right to you?

### Filesystem
```
/var/www/html
├── shared
│   └── web
|       └── volumes -> /var/www/html/web/volumes
└── web
    ├── cpresources
    ├── index.php
    ├── volumes
    │   └── files/
    │       ├── some-asset.jpg
    │       └── another-asset.jpg
    └── web.config
```

### Filesystem Config

![image](https://user-images.githubusercontent.com/1843073/206017087-3260eccb-83b5-46e6-b07c-c6ceb002d1bd.png)

### Volume Config
![image](https://user-images.githubusercontent.com/1843073/206017292-aab5c9df-d9d8-4a88-b6b9-b247c096178e.png)

We also [have some code](https://github.com/craftcms/cms/blob/ddb1d63e4ee55d21b61aedf68e3088dcc714d510/src/fs/Local.php#L145-L146) that will account for symlinks in there, but that only applies to the local filesystem type. I'm assuming that's the type of filesystem you're using, but it doesn't look like that was included in the original message.

**Comment by ThomasDeMarez** [comment]
Seems like you got the symlink the other way around.

### Filesystem 

```
/var/www
├── shared
│   └── web
│       └── volumes
│           └── files
│               ├── some-asset.jpg
│               └── some-other-asset.pdf
└── current
    ├── cpresources
    ├── index.php
    ├── volumes -> /var/www/html/volumes
    └── web.config
```

Maybe the code you referenced doesn't recognise the symlink because the actual path referenced in the filesystem is the files folder within the symlink and not the volumes symlink itself.

**Comment by ThomasDeMarez** [comment]
Found it, the code you linked in you previous reply is non existent in the [Craft CMS 4.3.1 release](https://github.com/craftcms/cms/blob/4.3.1/src/fs/Local.php#L142).

This was released in [4.3.2](https://github.com/craftcms/cms/blob/4.3.4/CHANGELOG.md#432---2022-11-16);
- Fixed asset indexing for Local filesystems that were configured to use a symlinked server path.

Upgrading Craft to this version fixed the problem.
It also keeps working even when the `@webroot` alias is not explicitly set.

Thanks for looking into this!
